libs/chain-signatures/contract/src/lib.rs

```diff
@@ -73,17 +73,25 @@ pub struct MpcContract {
     pending_requests: LookupMap<SignatureRequest, YieldIndex>,
     proposed_updates: ProposedUpdates,
     config: Config,
+    allowed_code_hashes: Vec<CodeHash>,
+    historical_code_hashes: Vec<CodeHash>,
+    upgrade_deadline: Option<BlockHeight>,
 }
+
 impl From<v0_state::MpcContractV0> for MpcContract {
     fn from(value: v0_state::MpcContractV0) -> Self {
-        MpcContract {
+        Self {
             protocol_state: value.protocol_state.into(),
             pending_requests: value.pending_requests,
             proposed_updates: ProposedUpdates::default(),
             config: value.config,
+            allowed_code_hashes: vec![],
+            historical_code_hashes: vec![],
+            upgrade_deadline: Some(TEE_UPGRADE_PERIOD),
```

Comment:
I think this is not correct. `upgrade_deadline` should be either set to `None`, meaning that there is no pending upgrade, or to `env::block_index() + TEE_UPGRADE_PERIOD`, meaning that there is a pending upgrade with a specific deadline

Reply:
Fixed in https://github.com/Near-One/mpc/pull/376/commits/90db97bc661b0ce7ccc8f08ad9e8687f44b0178b.